### ISSUE TITLE
Handle index updates after schema change.

### ISSUE THREAD

**Issue by BugenZhao** [risingwavelabs/risingwave#8448]

**Comment by chenzl25** [comment]
Just handling adding columns and dropping columns for auto-covered indexes seems more reasonable. For normal indexes, we can follow the PostgreSQL behavior.

**Comment by BugenZhao** [comment]
> Just handling adding columns and dropping columns for auto-covered indexes seems more reasonable.

Agree, though it needs some extra work. I guess in this case we should add a flag for the index catalog to mark whether it's auto covered?

**Comment by liurenjie1024** [comment]
> For ADD COLUMN: Currently, we don't have a field to indicate whether an index is created with or without an explicit include keyword, so I think it is acceptable to keep them the same. If we want to support auto-covered indexes, we must add a flag to the IndexCatalog.

Agree.

> For DROP COLUMN: it seems to be a dangerous operation to drop indexes implicitly if they include the dropped column. Telling users to check their indexes if some indexes include the dropped column looks safer. We can provide a global variable to decide whether to drop indexes including the dropped column if users want to change the default behavior.

I think in our current implementation, index is just another mv, so if we deny the request which may affect downstream mv, we should also deny this request which may affect index. But we should have a way to allow user to force drop mv/index, maybe cascades keyword is a good option, even if it's not compatible with pg.
